### DISCUSSION TITLE
Difference between ruff and isort outputs.

### DISCUSSION THREAD

**Discussion by vamshiaruru-virgodesigns** [astral-sh/ruff#3624] [Q&A] [answered]
Hi everyone, checking out the project for the first time. Firstly, great project! The auto fix is great!

I am confused about ruff's output for import sorting. For a small testing file like this
```python
from api.settings import config
from typing_extensions import ParamSpec
import redis as sync_redis
import redis.asyncio as redis

print(sync_redis, redis, ParamSpec, config)
```

Ruff outputs
```python
import redis as sync_redis
import redis.asyncio as redis
from api.settings import config
from typing_extensions import ParamSpec

print(sync_redis, redis, ParamSpec, config)
```

Where as isort outputs
```python
import redis as sync_redis
import redis.asyncio as redis
from typing_extensions import ParamSpec

from api.settings import config

print(sync_redis, redis, ParamSpec, config)
```

I find isort's output more appealing, but regardless of what I personally feel, I'd like to understand why ruff sorts the imports this way. What are the rules rust follows? How do they differ from the rules isort follows? Any help is appreciated, thank you.

**Comment by JonathanPlasse** [comment, marked as answer]
Can you try adding [`src`](https://beta.ruff.rs/docs/settings/#src) to your configuration?
```toml
src = ["api"]
# or if you use src layout
src = ["src"]
```

**Reply by vamshiaruru-virgodesigns** [reply]
Oh yes, thank you. After setting that, isort and ruff are behaving the same way. I must have missed that in the documentation. Thanks!

**Reply by JonathanPlasse** [reply]
You are welcome :slightly_smiling_face:
